Sources/SpeziHealthKitBulkExport/BulkExportSession.swift

```diff
@@ -0,0 +1,241 @@
+                        // if there is at least one not-to-be-skipped batch, bring it to the front and continue the loop, to handle it next.
+                        await MainActor.run {
+                            self.descriptor.pendingBatches.swapAt(idx, self.descriptor.pendingBatches.startIndex)
+                        }
+                        continue loop
+                    } else {
+                        break loop
+                    }
+                case nil:
+                    // the batch hasn't run yet. we simply continue with the code below
+                    let result: Processor.Output
+                    do {
+                        await MainActor.run {
+                            self.progress = .init(
+                                currentBatchIdx: numCompletedBatches + 1, // +1 bc we want it to be user-displayable, ie starting at 1.
+                                numTotalBatches: numTotalBatches,
+                                currentBatchDescription: batch.userDisplayedDescription
+                            )
+                        }
+                        result = try await self.queryAndProcess(sampleType: batch.sampleType, for: batch.timeRange)
+                    } catch let error as QueryAndProcessError {
+                        logger.error(
+                            "Failed to query and process batch \(String(describing: batch)): \(String(describing: error)). Will schedule for retry on next app launch."
+                        )
+                        await popBatch(withResult: .failure(error))
+                        continue loop
+                    } catch {
+                        // SAFETY: this is in fact unreachable: the `queryAndProcess` call above has a typed throw, but the compiler doesn't seem to understand this.
+                        fatalError("unreachable")
+                    }
+                    await batchResultHandler(result)
+                    await MainActor.run {
+                        popBatch(withResult: .success(()))
+                        numCompletedBatches += 1
+                    }
+                }
+            }
+            await MainActor.run {
+                self.task = nil
+                if self.state != .paused {
+                    self.state = .done
+                }
+            }
+        }
+    }
+    
+    
+    @MainActor
+    public func pause() { // swiftlint:disable:this missing_docs
+        switch state {
+        case .paused, .done:
+            return
+        case .running:
+            break
+        }
+        task?.cancel()
+    }
+}
+
+
+// NOTE: in Swift 6.0.3, we need to place this enum in the global scope, since nesting it in `BulkHealthExporter.ExportSession`
+// (where it belongs) will cause the compiler to crash.
+// Whatever the actual underlying issue here is, it seems to be fixed in Swift 6.1, so we'll hopefully be able to move this type soon.
```

Comment:
IMO we could put the Swift Package version to 6.1 here?

Reply:
we don't yet have Xcode 16.3; also, nesting this is probably no longer really important since we now have a dedicated target for the exporter

Reply:
Sounds good, you are right. Not really a problem.